application/controllers/SurveyAdministrationController.php

```diff
@@ -2618,6 +2635,7 @@ public function changeTemplate($iSurveyID, $template, $aResults = null, $bReturn
             if (!empty($bReturn)) {
                 $aResults[$iSurveyID]['title'] = $survey->correct_relation_defaultlanguage->surveyls_title;
                 $aResults[$iSurveyID]['result'] = false;
+                $aResults[$iSurveyID]['error'] = gT("User does not have valid permissions");
```

Comment:
No : permission is already checked at line https://github.com/LimeSurvey/LimeSurvey/blob/743d8b74d16892dfee04fa32194c902813d8f586/application/controllers/SurveyAdministrationController.php#L2634

`errorSummary($osSurvey)` maybe ?

(and why we check `surveyactivation` here ?)

Reply:
> (and why we check `surveyactivation` here ?)

This was like this before this commit. Probably they are just reusing a permission :)

Reply:
> No : permission is already checked at line 

This one is dirfferent : _unknow reason_ ?

Reply:
No, when you reach that point (L2638), you are lacking permissions.

Reply:
Oh yes … in the { }

Sorry